servicetalk-http-netty/src/main/java/io/servicetalk/http/netty/RetryingHttpRequesterFilter.java

```diff
@@ -0,0 +1,693 @@
+
+        private int maxRetries = 3;
+
+        @Nullable
+        private Function<HttpResponseMetaData, HttpResponseException> responseMapper;
+
+        private BiFunction<HttpRequestMetaData, RetryableException, BackOffPolicy>
+                retryRetryableExceptions = (requestMetaData, e) -> ofInstant();
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, IOException, BackOffPolicy>
+                retryIdempotentRequests;
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, DelayedRetry, BackOffPolicy>
+                retryDelayedRetries;
+
+        @Nullable
+        private BiFunction<HttpRequestMetaData, Throwable, BackOffPolicy>
+                retryOther;
+
+        /**
+         * By default, automatic retries wait for the associated {@link LoadBalancer} to be ready before triggering a
+         * retry for requests. This behavior may add latency to requests till the time the load balancer is ready
+         * instead of failing fast. This method allows controlling that behavior.
+         *
+         * @param waitForLb Whether to wait for the {@link LoadBalancer} to be ready before retrying requests.
+         * @return {@code this}.
+         */
+        public Builder waitForLoadBalancer(final boolean waitForLb) {
+            this.waitForLb = waitForLb;
+            return this;
+        }
+
+        /**
+         * By default, fail a request if the last signal from the associated {@link ServiceDiscoverer} was an error.
+         * This method disables that behavior.
+         *
+         * @param ignoreSdErrors ignore {@link ServiceDiscoverer} errors when evaluating a request failure.
+         * @return {@code this}.
+         */
+        public Builder ignoreServiceDiscovererErrors(final boolean ignoreSdErrors) {
+            this.ignoreSdErrors = ignoreSdErrors;
+            return this;
+        }
+
+        /**
+         * Set the maximum number of allowed retry operations before giving up, applied as total max across all retry
+         * functions (see. {@link #retryDelayedRetries(BiFunction)}, {@link #retryIdempotentRequests(BiFunction)},
+         * {@link #retryRetryableExceptions(BiFunction)}, {@link #retryOther(BiFunction)}).
+         *
+         * @param maxRetries Maximum number of allowed retries before giving up
+         * @return {@code this}
+         */
+        public Builder maxTotalRetries(final int maxRetries) {
+            if (maxRetries <= 0) {
+                throw new IllegalArgumentException("maxRetries: " + maxRetries + " (expected: >0)");
+            }
+            this.maxRetries = maxRetries;
+            return this;
+        }
+
+        public Builder responseMapper(final Function<HttpResponseMetaData, HttpResponseException> mapper) {
+            this.responseMapper = requireNonNull(mapper);
+            return this;
+        }
+
+        /**
+         * The retrying-filter will evaluate for {@link RetryableException}s in the request flow.
```

Comment:
For all functions, consider clarifying that `ofNoRetries()` can be used to avoid retries

Reply:
Don't see a change for this

Reply:
It's missing from one function (added), but it's there for the rest.
> To disable retries you can return {@link BackOffPolicy#NO_RETRIES} from the {@code mapper}.